go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -877,3 +885,38 @@ func (bsnc *BaseSecondaryNetworkController) requireDHCP(pod *corev1.Pod) bool {
 		bsnc.IsPrimaryNetwork() &&
 		bsnc.TopologyType() == types.Layer2Topology
 }
+
+func (bsnc *BaseSecondaryNetworkController) updateLogicalSwitchPortsForLiveMigrationPods(pod *corev1.Pod, lsp *nbdb.LogicalSwitchPort, ops []ovsdb.Operation, nadName, switchName string) ([]ovsdb.Operation, error) {
+	kubevirtLiveMigrationStatus, err := kubevirt.DiscoverLiveMigrationStatus(bsnc.watchFactory, pod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to discover Live-migration status: %w", err)
+	}
+	if kubevirtLiveMigrationStatus != nil {
+		if pod.Name == kubevirtLiveMigrationStatus.TargetPod.Name {
+			lsps := []*nbdb.LogicalSwitchPort{lsp}
+			kubevirtLiveMigrationInReadyState := kubevirtLiveMigrationStatus.State == kubevirt.LiveMigrationTargetDomainReady
+			lsp.Enabled = ptr.To(kubevirtLiveMigrationInReadyState)
+			if kubevirtLiveMigrationInReadyState {
+				// closing the sourcePod lsp to ensure traffic goes to the now ready targetPod.
+				sourcePodLspName := bsnc.GetLogicalPortName(kubevirtLiveMigrationStatus.SourcePod, nadName)
+				sourcePodLsp := &nbdb.LogicalSwitchPort{Name: sourcePodLspName}
+				sourcePodLsp, err = libovsdbops.GetLogicalSwitchPort(bsnc.nbClient, sourcePodLsp)
+				if err != nil {
+					return nil, fmt.Errorf("cannot fetch switch port for node %s: %v", sourcePodLspName, err)
+				}
+				sourcePodLsp.Enabled = ptr.To(false)
+				lsps = append(lsps, sourcePodLsp)
```

Comment:
For localnet and IC this is wrong but for localnet with non IC it make sense.

Reply:
I don't follow. You'll need to elaborate.

Reply:
The code we have here right now at localnet on run when "pod" is locally scheduled and at localnet with IC the source pod LSP is at the other node.

For non IC is fine since we have just one OVN zone.

Reply:
how is this any different from layer2 ?

Reply:
OK so, IIUC, for layer2 you have: 
- 4 LSPs
  - 1 src LSP and 1 dst *remote* LSP in the **src** node
  - 1 *remote* src LSP and 1 dst LSP in the **target** node

While for localnet you have:
- 2 LSPs
  - 1 src LSP in the **src** node; 1 dst LSP in the **target** node

Reply:
> OK so, IIUC, for layer2 you have:
> 
> * 4 LSPs
>   
>   * 1 src LSP and 1 dst _remote_ LSP in the **src** node
>   * 1 _remote_ src LSP and 1 dst LSP in the **target** node
> 
> While for localnet you have:
> 
> * 2 LSPs
>   
>   * 1 src LSP in the **src** node; 1 dst LSP in the **target** node
> 
> And can you elaborate on the IC/non-IC complexity ?

on non-IC since there is just one zone, the source LSP is at the target LSP so the code at this PR work as expected.

but that is not the case at IC.

> 
> In IC IIUC you're saying the ovnkube-controller of the **src** node doesn't even know about the **target** LSP (which is located in a different zone).

More or less how is this PR is done, the update live migration function will not be called on the source node since the code is only called when the pod is scheduled locally and the source pod virt-launcher is not changed during live migration so no events so no reconcile.

meaning that we have to change this and trate the case for localnet as reconcile always independenlty of the pod being local or not.

Reply:
Agreed outside this PR to continue localnet on followup PR